src/core/manager.c

```diff
@@ -3463,6 +3464,34 @@ static void log_taint_string(Manager *m) {
                    "MESSAGE_ID=" SD_MESSAGE_TAINTED_STR);
 }
 
+static void send_notify_ready_via_creds(void) {
+        _cleanup_free_ char *cred = NULL, *address = NULL;
+        SocketAddress a;
+        int r;
+
+        cred = path_join(SYSTEM_CREDENTIALS_DIRECTORY, "notify_socket");
```

Comment:
hmm, instead of going via hardcoded path here, i think we should just rely on the fact that the code in src/core/import-creds.c actually sets CREDENTIALS_DIRECTORY env var after it is done. Which means you can just use read_credential() directly, it will to the right thing.

Reply:
That said, I wonder if we shouldn't extend import_credentials() to basically do something like this once it is done:

```c
read_credential("notify.socket", &buf, &buf_size);
setenv("NOTIFY_SOCKET", buf, 1);
```

(of course, with more error checking, and with checking that "buf" does not contain a NUL byte)

i.e. that way we first import all the smbios/fw-cfg/… bits and pieces into credentials and then we'd import the notify socket address from there into the NOTIFY_SOCKET env var.

Benefit: we actually just implement the same NOTIFY_SOCKET proto everywhere the same way, just the address gets propagated in differently at different places.